src/pages/index.js

```diff
@@ -22,128 +23,270 @@ import {
+  popupSelector: "#edit-avatar-modal",
+  submitHandler: handleAvatarFormSubmit,
+});
+
+// set popup event listeners
+imgagePopup.setEventListeners();
+profilePopup.setEventListeners();
+addCardPopup.setEventListeners();
+deleteCardPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+
+//// Instantiate API ////
+const api = new Api({
+  baseUrl: baseUrl,
+  headers: {
+    authorization: token,
+    "Content-Type": "application/json",
+  },
+});
+
+//// Instantiate UserInfo ////
+const user = new UserInfo({
+  profileNameSelector,
+  profileDescriptionSelector,
+  avatarSelector,
+});
+
+api
+  .getUserInfo()
+  .then((result) => {
+    console.log(result);
+    // process the result
+    user.setUserInfo({
+      name: result.name,
+      about: result.about,
+      avatar: result.avatar,
+    });
+  })
+  .catch(console.error);
 
+//// Render initial cards ////
+const cardsSection = new Section(
+  { items: [], renderer: createCard },
+  gallerySelector
+);
+
+api
+  .getInitialCards()
+  .then((result) => {
+    if (result.length === 0) {
+      // If no cards are in the database
+      // add initialCards to server
+      initialCards.forEach((card) => {
+        api.addNewCard(card);
+      });
+    }
+
+    result.forEach((item) => {
+      // create new card
+      const newCard = createCard(item, openImageModal);
+      // Add new card to begining of card gallery
+      cardsSection.addItem(newCard);
+    });
+
+    cardsSection.renderItems();
+  })
+  .catch(console.error);
+
+//// Functions ////
 function fillProfileForm() {
   const userInfo = user.getUserInfo();
-  profilePopup.setInputValues(userInfo);
 
-  // profileNameInput.value = userInfo.name;
-  // profileDescriptionInput.value = userInfo.description;
+  console.log(`Fill Profile Form: userInfo`);
+  console.log(userInfo);
+
+  // reformat data (change about property to description)
+  const updatedData = { name: userInfo.name, description: userInfo.about };
+  profilePopup.setInputValues(updatedData);
+}
+
+function openEditAvatarModal() {
+  editAvatarPopup.open();
 }
 
-// Button Handlers
 function openEditProfileModal() {
   fillProfileForm();
   formValidators[profileForm.getAttribute("id")].resetValidation();
 
   profilePopup.open();
 }
 
+function openImageModal(data) {
+  imgagePopup.open(data);
+}
+
 function openAddCardModal() {
   addCardPopup.open();
 }
 
-// Button Event Listeners
+function createCard(data) {
+  const cardElement = new Card(
+    data,
+    cardTemplate,
+    openImageModal,
+    openDeleteModal,
+    handleLikeButton
+  );
+
+  return cardElement.getView();
+}
+
+function openDeleteModal(currentCardId, currentCardElement) {
+  deleteCardPopup.open();
+  deleteCardPopup.currentCardId = currentCardId;
+  deleteCardPopup.currentCardElement = currentCardElement;
+}
+
+function handleCardDelete() {
+  return api
+    .deleteCard(deleteCardPopup.currentCardId)
+    .then(deleteCardPopup.currentCardElement.remove())
```

Comment:
`    .then(deleteCardPopup.currentCardElement.remove())
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.